db/model/generator.js

```diff
@@ -144,6 +143,45 @@ module.exports = function generator(seq, dataTypes) {
     },
 
     hooks: {
+
+      beforeCreate(inst /* , opts */) {
+        const gtName = inst.generatorTemplate.name;
+        const gtVersion = inst.generatorTemplate.version;
+        return seq.models.GeneratorTemplate.getGTMatchingNameVersion(gtName,
+          gtVersion).then((ret) => {
+            if (!ret) {
+              throw new ValidationError('The Generator Template with name:' +
+                `${gtName} and version: ${gtVersion} was not found`);
+            }
+
+            /*
+             * go through the context definition and encrypt the related context
+             * fields in the generator
+             */
+          });
+      }, // beforeCreate
+
+      beforeUpdate(inst /* , opts */) {
+        const gtName = inst.generatorTemplate.name;
```

Comment:
if beforeCreate and beforeUpdate have same code in them, extract that block of code to a separate fn and call it from both - better for code maintainability

Reply:
Yes! Will take care of this while doing the encryption story.